smart-contracts/contracts/ContentDirectory.sol

```diff
@@ -0,0 +1,475 @@
+  function setChannelInstancesLimit(uint64 _newLimit) public onlyCouncil whenNotPaused {
+    channelInstancesLimit = _newLimit;
+    emit ChannelInstancesLimitUpdated(_newLimit);
+  }
+
+  function setVideosPerChannelDefaultLimit(uint32 _newLimit) public onlyCouncil whenNotPaused {
+    require(
+      _newLimit <= videosPerChannelMaxLimit,
+      "videosPerChannelDefaultLimit cannot exceed current videosPerChannelMaxLimit"
+    );
+    videosPerChannelDefaultLimit = _newLimit;
+    emit VideosPerChannelDefaultLimitUpdated(_newLimit);
+  }
+
+  function setVideosPerChannelMaxLimit(uint32 _newLimit) public onlyCouncil whenNotPaused {
+    require(
+      _newLimit >= videosPerChannelDefaultLimit,
+      "videosPerChannelMaxLimit must be greater than or equal to videosPerChannelDefaultLimit"
+    );
+    videosPerChannelMaxLimit = _newLimit;
+    emit VideosPerChannelMaxLimitUpdated(_newLimit);
+  }
+
+  constructor(
+    RuntimeAddressProvider _provider,
+    MembershipBridge _membershipBridge,
+    ContentWorkingGroupBridge _contentWorkingGroupBridge
+  )
+    public
+    // TODO: The upgraded logic contract would take those as args:
+    // ChannelStorage _channelStorage,
+    // VideoStorage _videoStorage,
+    // CuratorGroupStorage _curatorGroupStorage
+    RuntimeManageable(_provider)
+  {
+    membershipBridge = _membershipBridge;
+    contentWorkingGroupBridge = _contentWorkingGroupBridge;
+    channelStorage = new ChannelStorage();
+    videoStorage = new VideoStorage();
+    curatorGroupStorage = new CuratorGroupStorage();
+  }
+
+  // Faciliates migration to new logic contract
+  function migrate(
+    address _newLogic,
+    address _newVideoStorage,
+    address _newChannelStorage,
+    address _newCuratorGroupStorage
+  ) public onlyCouncil {
+    _migrateStorage(address(videoStorage), _newVideoStorage, _newLogic);
+    _migrateStorage(address(channelStorage), _newChannelStorage, _newLogic);
+    _migrateStorage(address(curatorGroupStorage), _newCuratorGroupStorage, _newLogic);
+    emit Migrated(
+      _newLogic,
+      _newVideoStorage == address(0) ? address(videoStorage) : _newVideoStorage,
+      _newChannelStorage == address(0) ? address(channelStorage) : _newChannelStorage,
+      _newCuratorGroupStorage == address(0) ? address(curatorGroupStorage) : _newCuratorGroupStorage
+    );
+    selfdestruct(msg.sender);
+  }
+
+  function _migrateStorage(
+    address _oldStorageAddress,
+    address _newStorageAddress,
+    address _newLogicAddress
+  ) internal {
+    if (_newStorageAddress != address(0)) {
+      // TODO: Set upgraded storage contract owner
+    }
+    Ownable oldStorage = Ownable(_oldStorageAddress);
+    oldStorage.transferOwnership(_newLogicAddress);
+  }
+
+  // CHANNELS
+  using ChannelOwnershipDecoder for ChannelOwnership;
+
+  // Access/validation utils:
+  function _validateOwnership(ChannelOwnership memory _ownership) internal view returns (bool) {
```

Comment:
Nicely spotted. It was supposed to be used instead of `ownership.isValid` in `createChannel`/`updateChannelOwnership`. I'll fix it and add a related test case